contrib/TestHarness2/test_harness/run.py

```diff
@@ -263,7 +285,7 @@ def choose_binary(self, test_file: Path) -> Path:
             max_version = Version.parse(version_expr[3])
         candidates: List[Path] = []
         for ver, binary in self.binaries.items():
-            if min_version <= ver <= max_version:
+            if min_version <= ver < max_version:
```

Comment:
The definition was exclusive upper bound when added,
so just change it back.
Though it's very rare to hit this.